src/xg.cpp

```diff
@@ -1585,11 +1416,19 @@ int64_t XG::rank_to_id(size_t rank) const {
         cerr << "[xg] error: Request for id of rank 0" << endl;
         assert(false);
     }
-    if(rank > i_iv.size()) {
-        cerr << "[xg] error: Request for id of rank " << rank << "/" << i_iv.size() << endl;
+    if(rank > node_count) {
+        cerr << "[xg] error: Request for id of rank " << rank << "/" << node_count << endl;
         assert(false);
     }
-    return i_iv[rank-1];
+    int i = 0;
```

Comment:
I think this is the source of the time outs. Won't this require you to iterate over the entire vector to get IDs at the end of it? That will give this algorithm linear time performance. Using the rank and select operations on g_bv, there should be a way to do it in constant time.